icu4c/source/common/uloc.cpp

```diff
@@ -1568,7 +1568,17 @@ _canonicalize(const char* localeID,
         if (localeID==NULL) {
            localeID=uloc_getDefault();
         }
-        tmpLocaleID=localeID;
+        if (uprv_stricmp(localeID, "root") == 0) {
+            tmpLocaleID = localeID + 4;
+        } else if (uprv_strnicmp(localeID, "und", 3) == 0 &&
+            (localeID[3] == '\0' ||
+             localeID[3] == '-' ||
```

Comment:
Please indent this line by 4 more than the previous one. Next two same level as this one.
(Or merge this + next 2 onto the line above.)